frontends/ast/simplify.cc

```diff
@@ -1981,11 +1981,20 @@ bool AstNode::simplify(bool const_fold, bool at_zero, bool in_lvalue, int stage,
 			varbuf->children[0] = buf;
 		}
 
-		if (type == AST_FOR) {
+		if (type == AST_FOR && init_ast->children[0]->id2ast->str.find("$fordecl_block") == std::string::npos) {
 			AstNode *buf = next_ast->clone();
 			delete buf->children[1];
 			buf->children[1] = varbuf->children[0]->clone();
 			current_block->children.insert(current_block->children.begin() + current_block_idx++, buf);
+		} else {
```

Comment:
I don't understand why we need to remove the variable from `current_scope`. Simply not generating the assignment seems like it should be enough. I double checked, and it seems the given test case does not require this portion of the changes to pass.

Reply:
It was done, to make sure we won't use this wire later in the code, but as it contains a unique prefix that prevents us from accessing it and opt_clean removes unused wires, it is true that we don't need this change.

I removed it.